lib/api/core/models/repositories/roleRepository.js

```diff
@@ -29,30 +29,39 @@ module.exports = function (kuzzle) {
       deferred = q.defer(),
       keys = [],
       buffer = {},
-      result = [];
+      result = [],
+      internalIndexAllowedIds = {},
+      restrictions = {};
 
-    roleKeys.forEach((roleKey) => {
+    roleKeys.forEach(roleKey => {
```

Comment:
Be careful, roleKeys should be an array of role Ids (strings) and not an array of role objects (sample usage here: https://github.com/kuzzleio/kuzzle/blob/53d744514c1f4e1eeebf023a68c77df06340c38f/lib/api/core/models/repositories/profileRepository.js#L134 )

as well, the notions of "internalIndexAllowed" and "restrictions" should not be handled at role repository's side, but rather at profile busniess object's side (those properties are dependant of the mapping btw. profile and role, not of the role himself).

Difficult to explain in a good english what I mean, but I can try to make a PR into your branch to be more clear ;)